src/lower/lowerer_impl.cpp

```diff
@@ -697,11 +697,15 @@ Stmt LowererImpl::lowerForall(Forall forall)
                                         getArgumentAccesses(forall),
                                         reducedAccesses);
 
-  // Emit temporary initialization if forall is sequential and leads to a where statement
+  // Emit temporary initialization if forall is sequential or parallelized by cpu threads and leads to a where statement
+  // This is for workspace hoisting by 1-level
   vector<Stmt> temporaryValuesInitFree = {Stmt(), Stmt()};
   auto temp = temporaryInitialization.find(forall);
   if (temp != temporaryInitialization.end() && forall.getParallelUnit() == ParallelUnit::NotParallel && !isScalar(temp->second.getTemporary().getType()))
     temporaryValuesInitFree = codeToInitializeTemporary(temp->second);
+  else if (temp != temporaryInitialization.end() && forall.getParallelUnit() == ParallelUnit::CPUThread && !isScalar(temp->second.getTemporary().getType())) {
```

Comment:
Some of these lines seem a bit long. Ideally, no line should be more than 80 characters long.

Reply:
I reduced the line sizes for lines: 700, 706, 2123, and 2126 in `src/lower/lowerer_impl.cpp`. 

All of the other lines in my changes seem to be within the stylistic limit of the github screen (80-100 characters).